tests/functional/adapter/test_python_submissions.py

```diff
@@ -1,89 +1,177 @@
+        return {"models": {"+incremental_strategy": "append"}}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": basic_sql, "incremental.py": incremental_python}
+
+    def test_incremental(self, project):
+        vars_dict = {
+            "test_run_schema": project.test_schema,
+        }
+
+        results = run_dbt(["run", "--vars", yaml.safe_dump(vars_dict)])
+        assert len(results) == 2
+
+
+class TestPythonClonePossible:
+    """Test that basic clone operations are possible on Python models. This
+    class has been adapted from the BaseClone and BaseClonePossible classes in
+    dbt-core and modified to test Python models in addition to SQL models."""
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": schema_yml,
+            "model.sql": basic_sql,
+            "my_python_model.py": basic_python,
+        }
+
+    @pytest.fixture(scope="class")
+    def other_schema(self, unique_schema):
+        return unique_schema + "_other"
+
+    @pytest.fixture(scope="class")
+    def profiles_config_update(self, dbt_profile_target, unique_schema, other_schema):
+        """Update the profiles config to duplicate the default schema to a
+        separate schema called `otherschema`."""
+        outputs = {"default": dbt_profile_target, "otherschema": deepcopy(dbt_profile_target)}
+        outputs["default"]["schema"] = unique_schema
+        outputs["otherschema"]["schema"] = other_schema
+        return {"test": {"outputs": outputs, "target": "default"}}
+
+    def copy_state(self, project_root):
+        """Copy the manifest.json project for a run into a separate `state/`
+        directory inside the project root, so that we can reference it
+        for cloning."""
+        state_path = os.path.join(project_root, "state")
+        if not os.path.exists(state_path):
+            os.makedirs(state_path)
+        shutil.copyfile(f"{project_root}/target/manifest.json", f"{state_path}/manifest.json")
+
+    def run_and_save_state(self, project_root):
+        """Run models and save the state to a separate directory to prepare
+        for testing clone operations."""
+        results = run_dbt(["run"])
+        assert len(results) == 2
+        self.copy_state(project_root)
+
+    def assert_relation_types_match_counter(self, project, schema, counter):
+        """Check that relation types in a given database and schema match the
+        counts specified by a Counter object."""
+        schema_relations = project.adapter.list_relations(database=project.database, schema=schema)
+        schema_types = [str(r.type) for r in schema_relations]
+        assert Counter(schema_types) == counter
+
+    def test_can_clone_true(self, project, unique_schema, other_schema):
+        """Test that Python models can be cloned using `dbt clone`. Adapted from
+        the BaseClonePossible.test_can_clone_true test in dbt-core."""
```

Comment:
Source for `BaseClonePossible.test_can_clone_true` here: https://github.com/dbt-labs/dbt-core/blob/d21ff38ba1aafdee67e0165856f59b31d824d144/tests/functional/adapter/dbt_clone/test_dbt_clone.py#L103-L137